eZ/Publish/Core/Persistence/Cache/ContentHandler.php

```diff
@@ -343,7 +343,8 @@ public function deleteVersion($contentId, $versionNo)
      */
     public function listVersions($contentId, $status = null, $limit = -1)
     {
-        $cacheItem = $this->cache->getItem("ez-content-${contentId}-version-list" . ($status ? "-byStatus-${status}" : '') . "-limit-{$limit}");
+        $cacheItem = $this->cache->getItem("ez-content-${contentId}-version-list" . ($status !== null ? "-byStatus-${status}" : '') . "-limit-{$limit}");
```

Comment:
+1, side note: logic here is being removed in https://github.com/ezsystems/ezpublish-kernel/pull/2840